New permissions to access smart rules [PAB-4929]

yta-c8y opened the pull request:
Will update the data and version before merging.

yta-c8y committed:
doc the new roles

yta-c8y committed:
Change log

yta-c8y committed:
Date

yta-c8y committed:
add Roles table

yta-c8y committed:
change to simple ticks

ben-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,14 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
 - The user has the required privileges to access Analytics Builder models and create new instances of Analytics builder templated models. Refer to [Permissions](/streaming-analytics/introduction-analytics/#permissions).
 
+#### Role-based Access
+Following Roles are reqiured to access Smart rules (NEW)
```
```suggestion
The following permissions are required to access Smart rules (NEW):
```

BeateRixen replied:
Change to "smart rules" as we decided not to capitalize this feature.

BeateRixen commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+---
+date: 2025-10-28
+title: Smart rules (NEW) introduces support for fine grained RBAC
```
```suggestion
title: Smart rules (NEW) introduces support for fine-grained RBAC
```

BeateRixen commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+version: 26.230.0
+---
+
+Smart rules (NEW) introduces support for fine grained RBAC.
```
```suggestion
The smart rules (NEW) feature has introduced support for fine-grained RBAC.
```

BeateRixen commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+
+Smart rules (NEW) introduces support for fine grained RBAC.
+
+For more details refer to [Role-based Access ](/streaming-analytics/smart-rules-plugin/#role-based-access).
```
```suggestion
For details, refer to [Role-based access](/streaming-analytics/smart-rules-plugin/#role-based-access).
```

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,14 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
 - The smart rules plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.
 - The user has the required privileges to access Analytics Builder models and create new instances of Analytics builder templated models. Refer to [Permissions](/streaming-analytics/introduction-analytics/#permissions).
 
+#### Role-based Access
```
```suggestion
#### Role-based access {#role-based-access}
```

yta-c8y committed:
Update content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

yta-c8y committed:
Update content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

yta-c8y committed:
Update content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

yta-c8y committed:
Update content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

yta-c8y committed:
Update content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

yta-c8y committed:
Permission not Roles

yta-c8y committed:
Mention the `Smart rule instances` shown in `Global roles` UI

yta-c8y committed:
linking Global Role

yta-c8y committed:
reverting deleted section

yta-c8y committed:
Merge branch 'develop' into dev/PAB-4929

skom-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,16 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+
+|privilege|Description|
+|---------|-----------|
+|**READ**|Grants view-only access to Smart rule instances. Users with this permission can view existing instances but cannot create, modify, or delete them.|
```
```suggestion
|**READ**|Grants read-only access to Smart rule instances. Users with this permission can view existing instances but cannot create, modify, or delete them.|
```

skom-c8y commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+version: 
+---
+
+The Smart rules (NEW) now supports fine-grained Role-Based Access Control (RBAC) to manage user permissions more precisely.
```
plugin?

yta-c8y replied:
Should I mention it explicitly as for plugin.

yta-c8y replied:
It would be too specific as we can also use these to update the templates from "Streaming analytics" as well, Though both are same. 
It would be better if we keep it as "Smart rules (NEW)".

skom-c8y commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+  - value: tc-KXXmo2SUR
+    label: apama-in-c8y
+ticket: PAB-4929
+version: 
```
26.247.0

skom-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,16 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+|---------|-----------|
+|**READ**|Grants view-only access to Smart rule instances. Users with this permission can view existing instances but cannot create, modify, or delete them.|
+|**ADMIN**|Grants full control over Smart rule instances. Users with this permission can Create, Read, Update, and Delete (CRUD) Smart rule instances.|
+
```
please add references to roles required to manage models

yta-c8y replied:
Have added description at the end of this section

skom-c8y commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+title: Smart rules (NEW) introduces support for fine-grained RBAC
+change_type:
+  - value: change-pXAlHAWka
+    label: Preview
```
Ideally, I would call this a `Feature` but since the current state of the plugin itself is in public preview, we can leave this in preview phase too

yta-c8y replied:
Ok leaving this as it is.

skom-c8y commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+---
+date: 2025-10-31
```
Don't forget to remove the date before you merge this

yta-c8y replied:
Will remove it before the final approval.

yta-c8y committed:
Apply suggestion from @skom-c8y

Co-authored-by: Sandeep Komarneni <87428699+skom-c8y@users.noreply.github.com>

yta-c8y committed:
Model permission reference in instances

yta-c8y committed:
updating link in change log

yta-c8y committed:
vesion numbere

yta-c8y committed:
Merge branch 'develop' into dev/PAB-4929

BeateRixen commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+version: 26.247.0
+---
+
+The Smart rules (NEW) now supports fine-grained Role-Based Access Control (RBAC) to manage user permissions more precisely.
```
```suggestion
The smart rules (NEW) feature now supports fine-grained Role-Based Access Control (RBAC) to manage user permissions more precisely.
```

BeateRixen commented on content/change-logs/analytics/apama-in-c8y-20251010-Role-based-access-to-Smart-rules.md:
```diff
@@ -0,0 +1,20 @@
+
+The Smart rules (NEW) now supports fine-grained Role-Based Access Control (RBAC) to manage user permissions more precisely.
+
+For details, refer to [Managing permissions to Smart rules (NEW) instances](/streaming-analytics/smart-rules-plugin/#managing-permissions-to-smart-rules-instances).
```
```suggestion
For details, refer to [Managing permissions to smart rules (NEW) instances](/streaming-analytics/smart-rules-plugin/#managing-permissions-to-smart-rules-instances).
```

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,20 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
 - The smart rules plugin is installed in the desired application, and the **Smart rules (NEW) plugin** is enabled in **Preview feature**.
 - The user has the required privileges to access Analytics Builder models and create new instances of Analytics builder templated models. Refer to [Permissions](/streaming-analytics/introduction-analytics/#permissions).
 
+#### Managing permissions to Smart rules (NEW) instances {#managing-permissions-to-smart-rules-instances}
```
```suggestion
#### Managing permissions to smart rules (NEW) instances {#managing-permissions-to-smart-rules-instances}
```

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,20 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+
+|privilege|Description|
+|---------|-----------|
+|**READ**|Grants read-only access to Smart rule instances. Users with this permission can view existing instances but cannot create, modify, or delete them.|
```
```suggestion
|**READ**|Grants read-only access to smart rule instances. Users with this permission can view existing instances but cannot create, update, or delete them.|
```

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,20 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+|privilege|Description|
+|---------|-----------|
+|**READ**|Grants read-only access to Smart rule instances. Users with this permission can view existing instances but cannot create, modify, or delete them.|
+|**ADMIN**|Grants full control over Smart rule instances. Users with this permission can Create, Read, Update, and Delete (CRUD) Smart rule instances.|
```
```suggestion
|**ADMIN**|Grants full control over smart rule instances. Users with this permission can create, read, update, and delete (CRUD) smart rule instances.|
```

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,20 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+
+**Smart rule instances** in [Global role](/standard-tenant/managing-permissions/#to-add-a-global-role) is provided to manage permissions required to access **Smart rules (NEW)** instances. 
+
+|privilege|Description|
```
```suggestion
|Permission type|Description|
```

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y committed:
Apply suggestion

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

skom-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,20 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+These permissions ensure fine-grained access control, allowing administrators to manage user privileges based on operational requirements.
+
+With this permission, the user can only view the model associated with the instance. The user is restricted from creating, editing, or deleting the underlying model. 
+These higher privileges are managed through the Analytics Builder [permissions](/streaming-analytics/introduction-analytics/#permissions), which are configured by the tenant administrator.
```
```suggestion
These higher privileges are managed through the Analytics Builder [permissions](/streaming-analytics/introduction-analytics/#permissions), which can be configured by the tenant administrator.
```

skom-c8y committed:
Apply suggestion

skom-c8y committed:
Clarify permissions for Analytics Builder access

Updated permissions section to clarify access levels for Analytics Builder models.

yta-c8y committed:
removing date will be added on release

niva-c8y reviewed (commented):
reads much better now, i just wonder about the terminology "admin", "management", and "full control"  and whether we could unify this a bit more to ensure that it reads fluently

BeateRixen commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,19 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
 
+#### Managing permissions to smart rules (NEW) instances {#managing-permissions-to-smart-rules-instances}
+
+**Smart rule instances** in [Global role](/standard-tenant/managing-permissions/#to-add-a-global-role) is provided to manage permissions required to access **Smart rules (NEW)** instances. 
```
```suggestion
The **Smart rule instances** permission under [global roles](/standard-tenant/managing-permissions/#to-add-a-global-role) allows you to manage permissions required to access smart rules (NEW) instances. 
```

ben-c8y committed:
Improve wording (from Beate)

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

ben-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,19 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+
+These permissions ensure fine-grained access control, allowing administrators to manage user privileges based on operational requirements.
+
+The permissions described above grant read-only access to the underlying Analytics Builder model. Management actions on the Analytics Builder model (like create, edit, or delete) require the *CEP management* roles, see [permissions](/streaming-analytics/introduction-analytics/#permissions).
```
```suggestion
The permissions described above grant read-only access to the underlying Analytics Builder model. Management actions on the Analytics Builder model (like create, edit, or delete) require the *CEP management* permissions, see [permissions](/streaming-analytics/introduction-analytics/#permissions).
```

ben-c8y commented on content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md:
```diff
@@ -16,6 +16,19 @@ Before using the smart rules (NEW) plugin, ensure that the following requirement
+
+The **Smart rule instances** permission under [global roles](/standard-tenant/managing-permissions/#to-add-a-global-role) allows you to manage permissions required to access smart rules (NEW) instances. 
+
+|Permission type|Description|
```
```suggestion
|Permission|Description|
```


Actually, "Smart rule instances" is the permission type, and READ/ADMIN are the specific permissions within that type

BeateRixen replied:
@ben-c8y I´m actually currently discussing this with @beabehnisch as I also stumbled over it here. That is the way I used it in the docs so far. So let´s take your suggestion for now. In general, we actually use "permission" for both, permission type and access rights.

yta-c8y committed:
Update content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

yta-c8y committed:
Update content/streaming-analytics/smart-rules-plugin-bundle/getting-started.md

Co-authored-by: Ben Spiller <84027153+ben-c8y@users.noreply.github.com>

yta-c8y committed:
unify terminology for admin, full control,management

yta-c8y committed:
grammer correctio

yta-c8y committed:
Merge branch 'develop' into dev/PAB-4929
